public/app/features/dashboard-scene/serialization/transformSaveModelToScene.ts

```diff
@@ -501,15 +501,18 @@ export function buildGridItemForPanel(panel: PanelModel): SceneGridItemLike {
     // To be replaced with it's own option persited option instead derived
     hoverHeader: !panel.title && !panel.timeFrom && !panel.timeShift,
     $data: createPanelDataProvider(panel),
-    menu: new VizPanelMenu({
-      $behaviors: [panelMenuBehavior],
-    }),
     titleItems,
 
     extendPanelContext: setDashboardPanelContext,
     _UNSAFE_customMigrationHandler: getAngularPanelMigrationHandler(panel),
   };
 
+  if (!config.publicDashboardAccessToken) {
+    vizPanelState.menu = new VizPanelMenu({
+      $behaviors: [panelMenuBehavior],
+    });
+  }
```

Comment:
@juanicabanas i think we need to disable keyboard shotcuts in the public dashboards too. They are currently active on a public dashboard, but i think it does not make sense, right?